### PR TITLE
DAOS-3794 test: Adding test to destroy pool during rebuild

### PR DESCRIPTION
**Pull request by saurabhtandan**
Adding test to destroy pool during rebuild after some
io has been performed.
After the pool is destroyed, try to re-create pool
on remaining ranks.

### CODE REVIEW

**Review by shimizukko — APPROVED** [review]
LGTM

**Comment by dinghwah** [comment]
The issue of skipForTicket DAOS-3794: rebuild going on and on after one rank is killed.
It seems occurred before this testcase (delete a pool during rebuild) started, 
I would suggest to have the pre-condition issue fixed and commit the new-test.

**Comment by saurabhtandan** [comment]
> The issue of skipForTicket DAOS-3794: rebuild going on and on after one rank is killed.
> It seems occurred before this testcase (delete a pool during rebuild) started,
> I would suggest to have the pre-condition issue fixed and commit the new-test.

As per conversation with Scott K, if the test is automated and there was a ticket opened associated to it, the test case can be merged with a skip decorator as long as a parallel ticket is opened to re-enable the skipped test and associated with the original issue resolving the problem.

File: src/tests/ftest/rebuild/pool_destroy_rebuild.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  name: daos_server
+  servers:
+    0:
+      targets: 4
+      pinned_numa_node: 0
+      nr_xs_helpers: 1
+      fabric_iface: ib0
+      fabric_iface_port: 31317
+      log_file: daos_server0.log
+      bdev_class: nvme
```
Comment (dinghwah) [inline src/tests/ftest/rebuild/pool_destroy_rebuild.yaml:21]: If it is possible to remove the nvme config on this test? If nvme is not needed in the rebuild test.

File: src/tests/ftest/rebuild/pool_destroy_rebuild.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+# change host names to your reserved nodes, the
+# required quantity is indicated by the placeholders
+hosts:
+  test_servers:
+    - server-A
+    - server-B
```
Comment (dinghwah) [inline src/tests/ftest/rebuild/pool_destroy_rebuild.yaml:6]: I think it needs more than 2 servers for the rebuild tests, specially rebuilding the rank-3 server.

File: src/tests/ftest/rebuild/pool_destroy_rebuild.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      fabric_iface_port: 31417
+      log_file: daos_server1.log
+      bdev_class: nvme
+      bdev_list: ["0000:da:00.0"]
+      scm_class: dcpm
+      scm_list: ["/dev/pmem1"]
+      scm_mount: /mnt/daos1
+testparams:
+  ranks:
+    rank_to_kill: 3
```
Comment (dinghwah) [inline src/tests/ftest/rebuild/pool_destroy_rebuild.yaml:40]: How can we rebuild rank3 with 2 servers?

**Review by dinghwah — CHANGES_REQUESTED** [review]

File: src/tests/ftest/rebuild/pool_destroy_rebuild.yaml (hunk 2 above)
Comment (saurabhtandan) [reply src/tests/ftest/rebuild/pool_destroy_rebuild.yaml:6]: There are more than one server started on each node. In total there will be 4 server ranks as there is servers_per_host: 2

File: src/tests/ftest/rebuild/pool_destroy_rebuild.yaml (hunk 1 above)
Comment (saurabhtandan) [reply src/tests/ftest/rebuild/pool_destroy_rebuild.yaml:21]: Any specific reason you want me to remove nvme? Whole intention is, if I am using hardware nodes anyways then why not make use of all the perks offered by it.

File: src/tests/ftest/rebuild/pool_destroy_rebuild.yaml (hunk 3 above)
Comment (saurabhtandan) [reply src/tests/ftest/rebuild/pool_destroy_rebuild.yaml:40]: Answered above

File: src/tests/ftest/rebuild/pool_destroy_rebuild.yaml (hunk 1 above)
Comment (dinghwah) [reply src/tests/ftest/rebuild/pool_destroy_rebuild.yaml:21]: Only a small number of hosts has nvme, we should remove the dependency to allow the rebuild functional test launch easier. 
On the rebuild functional test-dir, we should have all the scripts consist with similar test-environments and requirement.

File: src/tests/ftest/rebuild/pool_destroy_rebuild.yaml (hunk 2 above)
Comment (dinghwah) [reply src/tests/ftest/rebuild/pool_destroy_rebuild.yaml:6]: Thanks for the explanation. All of the current rebuild tests are on single server per host, rebuild on multiple-servers on the same host might causing DAOS-5868. 
We might need to add more tests for just restart server on the same host without the condition (pool destroy).

File: src/tests/ftest/rebuild/pool_destroy_rebuild.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+        # Wait for rebuild to start. If True just wait for rebuild to start,
+        # if False, wait for rebuild to complete.
+        self.pool.wait_for_rebuild(True, interval=1)
+        self.pool.set_query_data()
+
+        # destroy pool during rebuild
+        self.pool.destroy()
+        self.container = None
+
```
Comment (mjean308) [inline src/tests/ftest/rebuild/pool_destroy_rebuild.py:96]: Do you need to verify that the rebuild has not completed before destroying the pool?   i.e. rebuild['status']=busy

Comment (saurabhtandan) [reply src/tests/ftest/rebuild/pool_destroy_rebuild.py:99]: Done

**Comment by shimizukko** [comment]
Sorry. I'm on vacation, so not sure when I can do the review.

**Review by mjean308 — APPROVED** [review]
LGTM

**Review by shimizukko — APPROVED** [review]
LGTM

**Review by dinghwah — APPROVED** [review]
Since all tests passed, and stable.

**Pull request merged**